cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -185,26 +185,28 @@ func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticP
 		}
 	}
 
-	// ensure etcd certs are generated for etcd and kube-apiserver
-	if component == constants.Etcd || component == constants.KubeAPIServer {
-		if err := certsphase.CreateEtcdCACertAndKeyFiles(cfg); err != nil {
-			return fmt.Errorf("failed to upgrade the %s CA certificate and key: %v", constants.Etcd, err)
-		}
-	}
-	if component == constants.Etcd {
-		if err := certsphase.CreateEtcdServerCertAndKeyFiles(cfg); err != nil {
-			return fmt.Errorf("failed to upgrade the %s certificate and key: %v", constants.Etcd, err)
-		}
-		if err := certsphase.CreateEtcdPeerCertAndKeyFiles(cfg); err != nil {
-			return fmt.Errorf("failed to upgrade the %s peer certificate and key: %v", constants.Etcd, err)
+	if cfg.Etcd.Local != nil {
+		// ensure etcd certs are generated for etcd and kube-apiserver
+		if component == constants.Etcd || component == constants.KubeAPIServer {
+			if err := certsphase.CreateEtcdCACertAndKeyFiles(cfg); err != nil {
+				return fmt.Errorf("failed to upgrade the %s CA certificate and key: %v", constants.Etcd, err)
+			}
 		}
-		if err := certsphase.CreateEtcdHealthcheckClientCertAndKeyFiles(cfg); err != nil {
-			return fmt.Errorf("failed to upgrade the %s healthcheck certificate and key: %v", constants.Etcd, err)
+		if component == constants.Etcd {
+			if err := certsphase.CreateEtcdServerCertAndKeyFiles(cfg); err != nil {
+				return fmt.Errorf("failed to upgrade the %s certificate and key: %v", constants.Etcd, err)
```

Comment:
add the word "server" to this error message

Reply:
Sorry but I don't understand how the message should be changed :-(

Reply:
```
failed to upgrade the %s server certificate and key
```